reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Repeat.kt

```diff
@@ -3,19 +3,21 @@ package com.badoo.reaktive.observable
 import com.badoo.reaktive.base.ErrorCallback
 import com.badoo.reaktive.base.ValueCallback
 import com.badoo.reaktive.disposable.Disposable
-import com.badoo.reaktive.utils.atomic.AtomicInt
+import com.badoo.reaktive.utils.atomic.AtomicLong
 import com.badoo.reaktive.utils.serializer.serializer
 
 /**
- * Returns an [Observable] that automatically resubscribes to this [Observable] at most [count] times.
+ * Returns an [Observable] that automatically resubscribes to this [Observable] at most [times] times.
  *
  * Please refer to the corresponding RxJava [document](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#repeat-long-).
  */
-fun <T> Observable<T>.repeat(count: Int = -1): Observable<T> =
-    observable { emitter ->
+fun <T> Observable<T>.repeat(times: Long = Long.MAX_VALUE): Observable<T> {
+    require(times >= 0L) { "Number of times must not be negative" }
+
+    return observable { emitter ->
         val observer =
             object : ObservableObserver<T>, ValueCallback<T> by emitter, ErrorCallback by emitter {
-                private val counter: AtomicInt? = if (count >= 0) AtomicInt(count) else null
+                private val counter: AtomicLong? = if (times >= 0) AtomicLong(times) else null
```

Comment:
The condition `times >= 0` is always `true`. Looks like it should now be `if (times < Long.MAX_VALUE) AtomicLong(times) else null`?

Reply:
Fixed